web/ts/TUMLiveVjs.ts

```diff
@@ -210,23 +210,29 @@ export const watchProgress = function (streamID: number, lastProgress: number) {
         let iOSReady = false;
         let intervalMillis = 10000;
 
-        // Fetch the user's video progress from the database and set the time in the player
-        player.on("loadedmetadata", () => {
-            duration = player.duration();
-            player.currentTime(lastProgress * duration);
-        });
-
-        // iPhone/iPad need to set the progress again when they actually play the video. That's why loadedmetadata is
-        // not sufficient here.
-        // See https://stackoverflow.com/questions/28823567/how-to-set-currenttime-in-video-js-in-safari-for-ios.
-        if (videojs.browser.IS_IOS) {
-            player.on("canplaythrough", () => {
-                // Can be executed multiple times during playback
-                if (!iOSReady) {
-                    player.currentTime(lastProgress * duration);
-                    iOSReady = true;
-                }
+        // check if query contains parameter 't' and set timestamp accordingly
+        const queryT = getQueryParam("t");
+        if (queryT !== undefined) {
+            player.currentTime(Number(queryT));
```

Comment:
This should probably include the same special handling as for iOS, which can be encapsulated in a function to be called from both places.